lib/child_process.js

```diff
@@ -276,31 +273,30 @@ ObjectDefineProperty(exec, promisify.custom, {
  *   ) => any} [callback]
  * @returns {ChildProcess}
  */
-function execFile(file /* , args, options, callback */) {
-  let args = [];
-  let callback;
-  let options;
-
-  // Parse the optional positional parameters.
-  let pos = 1;
-  if (pos < arguments.length && ArrayIsArray(arguments[pos])) {
-    args = arguments[pos++];
-  } else if (pos < arguments.length && arguments[pos] == null) {
-    pos++;
-  }
-
-  if (pos < arguments.length && typeof arguments[pos] === 'object') {
-    options = arguments[pos++];
-  } else if (pos < arguments.length && arguments[pos] == null) {
-    pos++;
+function execFile(file, args = [], options, callback) {
+  if (args == null) {
+    args = [];
+  } else if (typeof args === 'object') {
+    if (!ArrayIsArray(args)) {
```

Comment:
```suggestion
  if (typeof args === 'object' && !ArrayIsArray(args)) {
```

Reply:
With the way the PR is now, if `execFile('ls', null, {}, () => {})` works, as it should. With the change proposed here, it will throw because this `typeof null` is `'object'`, so this code treats `args` like it's an `options` object, and then assigns `callback` to the actual `options` object. When `callback` is subsequently verified to be a function, the code throws.

This is they the code here tests for `null` and `undefined` first.

Reply:
If we use `if (args != null && typeof args === 'object' && !ArrayIsArray(args)) {` instead, that seems to work. So I'll do that.